File: framework/languages/java/samples/java/ShoppingMall/Server/OrderWorkflow/src/main/java/systems/zlink/samples/shoppingmall/server/orderworkflow/domain/OrderEvents.java
package systems.zlink.samples.shoppingmall.server.orderworkflow.domain;

import java.util.List;

/**
 * Order domain events.
 *
 * <p>Each event type maps to its simple class name; the same names are used as
 * the {@code eventType} discriminator inside the event store and as the
 * sequence the self-check assertion compares against.
 */
public final class OrderEvents {
    private OrderEvents() {
    }

    public static final String OrderStarted = "OrderStartedEvent";
    public static final String InventoryReserved = "InventoryReservedEvent";
    public static final String InventoryReservationFailed = "InventoryReservationFailedEvent";
    public static final String PaymentAuthorized = "PaymentAuthorizedEvent";
    public static final String PaymentFailed = "PaymentFailedEvent";
    public static final String InventoryReleased = "InventoryReleasedEvent";
    public static final String OrderConfirmed = "OrderConfirmedEvent";
    public static final String OrderFailed = "OrderFailedEvent";

    public record OrderLine(String sku, int quantity) {
    }

    public record StartOrderCommand(
        String orderId,
        String cartId,
        String shippingAddressId,
        String paymentMethodId,
        String idempotencyKey,
        List<OrderLine> lines,
        double amount,
        String currency) {
    }

    public record OrderStartedEvent(
        String eventId,
        String sourceCommandId,
        String orderId,
        String cartId,
        String shippingAddressId,
        String paymentMethodId,
        List<OrderLine> lines,
        double amount,
        String currency,
        long createdAtUnixMs) {
    }

    public record InventoryReservedEvent(
        String eventId, String orderId, String reservationId, long createdAtUnixMs) {
    }

    public record InventoryReservationFailedEvent(
        String eventId, String orderId, String reason, long createdAtUnixMs) {
    }

    public record PaymentAuthorizedEvent(
        String eventId, String orderId, String paymentId, long createdAtUnixMs) {
    }

    public record PaymentFailedEvent(
        String eventId, String orderId, String reason, long createdAtUnixMs) {
    }

    public record InventoryReleasedEvent(
        String eventId, String orderId, String reservationId, String reason, long createdAtUnixMs) {
    }

    public record OrderConfirmedEvent(
        String eventId, String orderId, long confirmedAtUnixMs) {
    }

    public record OrderFailedEvent(
        String eventId, String orderId, String reason, long failedAtUnixMs) {
    }
}

File: framework/languages/java/samples/kotlin/ShoppingMall/Server/OrderWorkflow/src/main/kotlin/systems/zlink/samples/kotlin/shoppingmall/server/orderworkflow/domain/OrderEvents.kt
package systems.zlink.samples.kotlin.shoppingmall.server.orderworkflow.domain

import systems.zlink.samples.kotlin.shoppingmall.shared.contracts.OrderLineInput

/**
 * Order domain events. Each event type maps to its simple class name, which is
 * also the `eventType` discriminator in the store and the sequence the
 * self-check assertion compares against.
 */
object OrderEventTypes {
    const val OrderStarted = "OrderStartedEvent"
    const val InventoryReserved = "InventoryReservedEvent"
    const val InventoryReservationFailed = "InventoryReservationFailedEvent"
    const val PaymentAuthorized = "PaymentAuthorizedEvent"
    const val PaymentFailed = "PaymentFailedEvent"
    const val InventoryReleased = "InventoryReleasedEvent"
    const val OrderConfirmed = "OrderConfirmedEvent"
    const val OrderFailed = "OrderFailedEvent"
}

data class OrderStartedEvent(
    val eventId: String,
    val sourceCommandId: String,
    val orderId: String,
    val cartId: String,
    val shippingAddressId: String,
    val lines: List<OrderLineInput>,
    val amount: Double,
    val currency: String,
    val createdAtUnixMs: Long,
)

data class InventoryReservedEvent(
    val eventId: String,
    val orderId: String,
    val reservationId: String,
    val createdAtUnixMs: Long,
)

data class InventoryReservationFailedEvent(
    val eventId: String,
    val orderId: String,
    val reason: String,
    val createdAtUnixMs: Long,
)

data class PaymentAuthorizedEvent(
    val eventId: String,
    val orderId: String,
    val paymentId: String,
    val createdAtUnixMs: Long,
)

data class PaymentFailedEvent(
    val eventId: String,
    val orderId: String,
    val reason: String,
    val createdAtUnixMs: Long,
)

data class InventoryReleasedEvent(
    val eventId: String,
    val orderId: String,
    val reservationId: String,
    val reason: String,
    val createdAtUnixMs: Long,
)

data class OrderConfirmedEvent(
    val eventId: String,
    val orderId: String,
    val confirmedAtUnixMs: Long,
)

data class OrderFailedEvent(
    val eventId: String,
    val orderId: String,
    val reason: String,
    val failedAtUnixMs: Long,
)
